src/main/groovy/com/cloudcard/photoDownloader/OrigoStorageService.groovy

```diff
@@ -0,0 +1,134 @@
+package com.cloudcard.photoDownloader
+
+import org.springframework.beans.factory.annotation.Value
+
+import javax.annotation.PostConstruct
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Service
+
+import static com.cloudcard.photoDownloader.ApplicationPropertiesValidator.throwIfBlank
+import static com.cloudcard.photoDownloader.ApplicationPropertiesValidator.throwIfTrue
+
+@Service
+@ConditionalOnProperty(value = "downloader.storageService", havingValue = "OrigoStorageService")
+class OrigoStorageService implements StorageService {
+
+    static final Logger log = LoggerFactory.getLogger(OrigoStorageService.class)
+
+    @Autowired
+    @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
+    FileNameResolver fileNameResolver
+
+    @Autowired
+    OrigoClient origoClient
+
+    @Value('${Origo.overrideCurrentPhoto}')
+    boolean overrideCurrentPhoto
```

Comment:
It's a little pedantic, but I would rename this variable to replaceExistingPhotos and default it to true.

Reply:
I think I can pull some strings to make that happen.